c7n/filters/health.py

```diff
@@ -40,39 +41,51 @@ def process(self, resources, event=None):
             return resources
 
         client = local_session(self.manager.session_factory).client('health')
-        m = self.manager.get_model()
-
-        f = {'services': [m.service.upper()],
-             'eventStatusCodes': self.data.get(
-                 'statuses', ['open', 'upcoming'])}
-
-        if self.data.get('eventTypeCodes'):
-            f['eventTypeCodes'] = self.data.get('types')
-
-        resource_map = {r[m.id]: r for r in resources}
+        f = self.get_filter()
+        resource_map = {r[self.manager.get_model().id]: r for r in resources}
         found = set()
         seen = set()
 
         for resource_set in chunks(resource_map.keys(), 100):
             f['entityValues'] = resource_set
             events = client.describe_events(filter=f)['events']
             events = [e for e in events if e['arn'] not in seen]
+            entities = []
+            self.process_event(events, entities)
+
+            event_map = {e['arn']: e for e in events}
+            for e in entities:
+                rid = e['entityValue']
+                if rid not in resource_map:
+                    continue
+                resource_map[rid].setdefault(
+                    'c7n:HealthEvent', []).append(event_map[e['eventArn']])
+                found.add(rid)
+            seen.update(event_map.keys())
+        return [resource_map[rid] for rid in found]
 
-            for event_set in chunks(events, 10):
-                event_map = {e['arn']: e for e in event_set}
-                for d in client.describe_event_details(
-                        eventArns=event_map.keys()).get('successfulSet', ()):
-                    event_map[d['event']['arn']]['Description'] = d[
-                        'eventDescription']['latestDescription']
-                entities = client.describe_affected_entities(
-                    filter={'eventArns': event_map.keys()})['entities']
+    def get_filter(self):
+        m = self.manager
+        if m.data['resource'] == 'ebs':
+            service = 'EBS'
+        else:
+            service = m.get_model().service.upper()
+        f = {'services': [service],
+             'eventStatusCodes': self.data.get(
+                 'statuses', ['open', 'upcoming'])}
+        if self.data.get('types'):
+            f['eventTypeCodes'] = self.data.get('types')
+        return f
 
-                for e in entities:
-                    rid = e['entityValue']
-                    if rid not in resource_map:
-                        continue
-                    resource_map[rid].setdefault(
-                        'c7n:HealthEvent', []).append(event_map[e['eventArn']])
-                    found.add(rid)
-                seen.update(event_map.keys())
-        return [resource_map[rid] for rid in found]
+    def process_event(self, health_events, entities):
```

Comment:
process event should just return the entities instead of mutating a parameter, afaics all the callers just pass an empty list.